### ISSUE TITLE
Objects not being parsed

### ISSUE THREAD

**Issue by max-waters** [pellierd/pddl4j#15]
I just noticed some strange behaviour. This code:

```
Parser parser = new Parser();
parser.parse(new File("pddl/blocksworld/domain.pddl"), new File("pddl/blocksworld/p01.pddl"));
System.out.println(parser.getProblem());
```

Gives this output as expected:

```
(define (problem blocks-4-0)
(:domain blocks)
(:requirements)
(:objects 
  d - BLOCK
  b - BLOCK
  a - BLOCK
  c - BLOCK
)
(:init
  (clear c)
  (clear a)
  (clear b)
  (clear d)
  (notable c)
  (notable a)
  (notable b)
  (notable d)
  (hand empty
)
(:goal   (and (on d c)
  (on c b)
  (on b a)))
)
```

However, this code:

```
Parser parser = new Parser();
parser.parseDomain("pddl/blocksworld/domain.pddl");
parser.parseProblem(new File("pddl/blocksworld/p01.pddl"));
System.out.println(parser.getProblem());
```

Gives this output:

```
(define (problem blocks-4-0)
(:domain blocks)
(:requirements)
(:init
  (clear c)
  (clear a)
  (clear b)
  (clear d)
  (notable c)
  (notable a)
  (notable b)
  (notable d)
  (hand empty
)
(:goal   (and (on d c)
  (on c b)
  (on b a)))
)
```

Note that in the second example, the objects have not been parsed.

**Comment by pellierd** [comment]
The behavior you describe is a normal behavior. The problem and the domain are not complet when error occurs during parsing.

Add the line:
` parser.getErrorManager().printAll();`
at the end of your code you will see that there is no errors when you parser in the first case and errors otherwise.

The parsing errors occurs because you parse separately the domain and the problem. When parsing the problem, the parser does not know that typing are allowed. Thus it refuses to read the list of objects that are typed. If you want to remove this error you have to specify in the problem the requirement :typing or parser the domain and the problem together.

In conclusion, your both methods are not equivalent.
